src/app/datagrid/pagination-conditional/pagination-conditional.ts

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) 2016-2018 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+import {ChangeDetectorRef, Component, QueryList, ViewChild, ViewChildren} from "@angular/core";
+
+import {ClrDatagridPagination} from "../../../clr-angular/data/datagrid";
+import {Inventory} from "../inventory/inventory";
+import {User} from "../inventory/user";
+
+const EXAMPLE = `
+<-- Inside the full datagrid declaration -->
+<clr-dg-footer>
+    {{pagination.firstItem + 1}} - {{pagination.lastItem + 1}}
+    of {{pagination.totalItems}} users
+    <clr-dg-pagination #pagination [clrDgPageSize]="10"></clr-dg-pagination>
+</clr-dg-footer>
+`;
+
+@Component({
+    selector: "clr-datagrid-conditional-pagination-demo",
+    providers: [Inventory],
+    templateUrl: "pagination-conditional.html",
+    styleUrls: ["../datagrid.demo.scss"]
+})
+export class DatagridConditionalPaginationDemo {
+    example = EXAMPLE;
+    users: User[];
+    paginationEnabled: boolean = false;
+
+    constructor(private inventory: Inventory, private cdr: ChangeDetectorRef) {
```

Comment:
I don't understand why `ChangeDetectorRef` is used in the demo app. Doesn't it mean users also will face the chocolate error and the need to use `ChangeDetectorRef`?

Reply:
I need @youdz to weigh in on this. I believe required because when the user toggles the pagination, the `pagination` variable is updated. Once that is updated, I get a chocolate error. Its happening because `ViewChildren`/`ViewChild` populates when the view changes and when the view changes the change detection has already begun.